### PR TITLE
tidb-lightning,br: replaced black-white-list by table-filter (#3065)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #3065 to release-3.1

---

### What is changed, added or deleted? (Required)

This PR documented the "table filter" feature shared by all new TiDB ecosystem tools, superseding the old "black-white-list" feature. The documentation is heavily copied from https://github.com/pingcap/tidb-tools/blob/master/pkg/table-filter/README.md.

The feature is / will be available in

* Lightning (v4.0.2, v3.1.3, v3.0.17)
* BR (v4.0.0-rc, v3.1.0-rc)
* Dumpling
* CDC (v4.0.2)
* (DM may also switch to this in the future.)

The Lightning docs have been revised for this. The existing "black-white-list" page has been redirected to the shared reference. ("black-white-list" still works but are no longer recommended. Therefore we don't bother to rename these into "block-allow-list")

The BR docs have been expanded to describe `-f`.

Dumpling is untouched, but whoever translating pingcap/docs-cn#3541 should point the link in `--filter` to this new document rather than the README.md. 

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions) (only BR and Lightning)
- [x] v3.0 (TiDB 3.0 versions) (only Lightning)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

    This PR documents features included in

   * pingcap/tidb-lightning#332
   * pingcap/br#313

    Translator working on pingcap/docs-cn#3541 should be aware of this PR.

### CODE REVIEW

**Comment by ran-huang** [comment]
@kennytm PTAL at this commit https://github.com/pingcap/docs/pull/3139/commits/43c6e18fee5e01907d31a9f4bba6597e08a151e0 which involves some version split changes.

File: table-filter.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+summary: Usage of table filter feature in TiDB tools.
+category: reference
+aliases: ['/docs/v3.1/tidb-lightning/tidb-lightning-table-filter/','/docs/v3.1/reference/tools/tidb-lightning/table-filter/','/tidb/v3.1/tidb-lightning-table-filter/']
+---
+
+# Table Filter
+
+The TiDB ecosystem tools operate on all the databases by default, but oftentimes only a subset is needed. For example, you only want to work with the schemas in the form of `foo*` and `bar*` and nothing else.
+
+Since TiDB 4.0, all TiDB ecosystem tools share a common filter syntax to define subsets. This document describes how to use the table filter feature.
```
Comment (kennytm) [inline table-filter.md:12]: Again maybe not refer to 4.0 here.

Comment (ran-huang) [inline table-filter.md:12]:
```suggestion
Several TiDB ecosystem tools share a common filter syntax to define subsets. This document describes how to use the table filter feature.
```

**Pull request merged**